### PR TITLE
Fix domain error

### PR DESCRIPTION
**Pull request by andrey-p**
**Reviewer:** @dharb 
**CC:** @jdorweiler 

## Description:

Make sure domain names with `_` in them don't lead to a JS error.

## Steps to test this PR:

1. Open https://www.zive.cz/ while monitoring the logs from the background page
2. You shouldn't get any JS errors (without the fix you get an error for `https://tracker_cnc.contentexchange.me/widget/EszYqNuK3wrRyDXwX-cibtgexhoe`)
3. Go to the `test/index.html` page and run unit tests
4. The new test I added shouldn't fail (it fails without the fix)

Note, there's some other tests failing there. I'm not sure what that's about.

## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: shared/js/trackers.js [hunk 1]
```diff
@@ -46,7 +46,25 @@ require.scopes.trackers = (function() {
...
+            if (parsedUrl && parsedUrl.hostname) {
+                hostname = parsedUrl.hostname
+            } else {
+                // fail gracefully if tldjs chokes on the URL e.g. it doesn't parse
+                // if the subdomain name has underscores in it
+                try {
+                    // last ditch attempt to try and grab a hostname
+                    // this will fail on more complicated URLs with e.g. ports
+                    // but will allow us to block simple trackers with _ in the subdomains
+                    hostname = urlToCheck.match(/^(?:.*:\/\/)([^/]+)/)[1]
```
Comment (dharb) [inline shared/js/trackers.js]: Nice!

**Comment by dharb** [comment]
I think this should cover our bases for random characters appearing in subdomains. 👍

File: shared/js/trackers.js [hunk 2]
```diff
@@ -46,7 +46,25 @@ require.scopes.trackers = (function() {
         if (settings.getSetting('trackerBlockingEnabled')) {
 
             let parsedUrl = tldjs.parse(urlToCheck)
-            let urlSplit = parsedUrl.hostname.split('.')
+            let hostname
+
+            if (parsedUrl && parsedUrl.hostname) {
+                hostname = parsedUrl.hostname
```
Comment (MariagraziaAlastra) [inline shared/js/trackers.js]: @andrey-p Hey, not sure if you saw it, but we have a [util function to get the hostname](https://github.com/duckduckgo/duckduckgo-privacy-extension/blob/develop/shared/js/utils.js#L6) - is that something that could have been useful here?

Comment (andrey-p) [reply shared/js/trackers.js]: Ah, didn't realize that existed. It uses `tldjs` under the hood. Maybe we can consolidate the two.

**Pull request merged**